### PR TITLE
Win UI 3 Update and Enhanced Navigation

### PR DESCRIPTION
**Pull request by nick5616**
### **Overview of changes**
image

1. Update FactoryOrchestrator to ingest the Win UI 3 library
2. Modify the MainPage.xaml to use the WinUI implementation of NavigationView which allows navigation items in the accessibility component tree to be seen as TabItem, rather than ListItem
3. Modify the structure and layout of the main page to increase application usability

### Updated app header
image
The new app header layout features identifier information to the left, while more information and control items are on the right
image
Underneath the updated header is the updated NavigationView, which is visualized in the following figure.
image
### Before and After (Dark mode)
**Before**
image
**After**
image
#

### CODE REVIEW

File: src/UWPClientLibrary/Microsoft.FactoryOrchestrator.UWPClient.csproj [hunk 1]
```diff
@@ -23,7 +22,7 @@
     <TargetName>Microsoft.FactoryOrchestrator.UWPClient</TargetName>
     <DefaultLanguage>en-US</DefaultLanguage>
     <TargetPlatformIdentifier>UAP</TargetPlatformIdentifier>
-    <TargetPlatformVersion Condition=" '$(TargetPlatformVersion)' == '' ">10.0.17763.0</TargetPlatformVersion>
+    <TargetPlatformVersion Condition=" '$(TargetPlatformVersion)' == '' ">10.0.19041.0</TargetPlatformVersion>
     <TargetPlatformMinVersion>10.0.17763.0</TargetPlatformMinVersion>
```
Comment (patelp3989) [inline src/UWPClientLibrary/Microsoft.FactoryOrchestrator.UWPClient.csproj:26]: Should we update the minimum version  ?

File: src/App/SaveLoadEditPage.xaml [hunk 2]
```diff
@@ -7,7 +7,7 @@
     xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
     xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
     mc:Ignorable="d"
-    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
```
Comment (patelp3989) [inline src/App/SaveLoadEditPage.xaml:10]: Any specific reason to remove this ?
Other pages have this as well. If we are removing it we should do it for other pages as well just for consistency.

File: src/App/MainPage.xaml.cs [hunk 3]
```diff
@@ -523,6 +525,16 @@ public void Dispose()
         private List<Tuple<string, string>> ipAddresses;
         private readonly SemaphoreSlim ipAddressSem;
         private readonly ResourceLoader resourceLoader = ResourceLoader.GetForCurrentView();
+
+        private void NavView_Loaded(object sender, RoutedEventArgs e)
```
Comment (patelp3989) [inline src/App/MainPage.xaml.cs:529]: Empty function.

File: src/App/MainPage.xaml.cs [hunk 4]
```diff
@@ -523,6 +525,16 @@ public void Dispose()
         private List<Tuple<string, string>> ipAddresses;
         private readonly SemaphoreSlim ipAddressSem;
         private readonly ResourceLoader resourceLoader = ResourceLoader.GetForCurrentView();
+
+        private void NavView_Loaded(object sender, RoutedEventArgs e)
+        {
+
+        }
+
+        private void NavView_ItemInvoked_1(MUXC.NavigationView sender, MUXC.NavigationViewItemInvokedEventArgs args)
```
Comment (patelp3989) [inline src/App/MainPage.xaml.cs:534]: Empty function.

File: src/App/AboutPage.xaml [hunk 5]
```diff
@@ -6,7 +6,7 @@
     xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
     xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
     mc:Ignorable="d"
-    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
```
Comment (patelp3989) [inline src/App/AboutPage.xaml:9]: Any specific reason to remove this ?

Comment (nick5616) [reply src/App/AboutPage.xaml:9]: Due to the way SystemResources color components in WinUI 3, adding a background brush as so would create an off-color rectangle in the middle of the frame. It looks very out of place.

**Comment by spaceisfun** [comment]
👀

File: src/App/MainPage.xaml.cs [hunk 6]
```diff
@@ -1,6 +1,7 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
+using Microsoft.UI.Xaml.Controls;
```
Comment (spaceisfun) [inline src/App/MainPage.xaml.cs:4]: do you need both this and the one on line 19?

File: src/UWPClientLibrary/Microsoft.FactoryOrchestrator.UWPClient.csproj (hunk 1 above)
Comment (nick5616) [reply src/UWPClientLibrary/Microsoft.FactoryOrchestrator.UWPClient.csproj:26]: WinUI 3 highly encourages updating the target platform version to build 19041. As for the minimum version, they recommend build 17763 which is what we already have

File: src/App/MainPage.xaml.cs [hunk 7]
```diff
@@ -15,6 +16,7 @@
 using Windows.UI.Xaml.Controls;
```
Comment (spaceisfun) [inline src/App/MainPage.xaml.cs:16]: is Windows.UI.Xaml.Controls still used? if not, you could do "using Controls = Microsoft.UI.Xaml.Controls;" i think

**Review by spaceisfun — DISMISSED** [review]

**Review by patelp3989 — APPROVED** [review]

File: src/App/MainPage.xaml.cs (hunk 6 above)
Comment (nick5616) [reply src/App/MainPage.xaml.cs]: No I do not! It's been removed

File: src/App/MainPage.xaml.cs (hunk 7 above)
Comment (nick5616) [reply src/App/MainPage.xaml.cs:15]: That namespace is used everywhere, unfortunately

**Review by nick5616 — COMMENTED** [review]
Looks good!

**Review by patelp3989 — DISMISSED** [review]

File: src/App/MainPage.xaml.cs (hunk 7 above)
Comment (spaceisfun) [reply src/App/MainPage.xaml.cs:15]: figured.

File: src/UWPClientLibrary/Microsoft.FactoryOrchestrator.UWPClient.csproj [hunk 8]
```diff
@@ -23,7 +22,7 @@
     <TargetName>Microsoft.FactoryOrchestrator.UWPClient</TargetName>
     <DefaultLanguage>en-US</DefaultLanguage>
     <TargetPlatformIdentifier>UAP</TargetPlatformIdentifier>
-    <TargetPlatformVersion Condition=" '$(TargetPlatformVersion)' == '' ">10.0.17763.0</TargetPlatformVersion>
+    <TargetPlatformVersion Condition=" '$(TargetPlatformVersion)' == '' ">10.0.19041.0</TargetPlatformVersion>
```
Comment (spaceisfun) [inline src/UWPClientLibrary/Microsoft.FactoryOrchestrator.UWPClient.csproj:25]: be sure to update the readme if it refers to 17763

**Review by spaceisfun — DISMISSED** [review]

File: src/UWPClientLibrary/Microsoft.FactoryOrchestrator.UWPClient.csproj (hunk 8 above)
Comment (spaceisfun) [reply src/UWPClientLibrary/Microsoft.FactoryOrchestrator.UWPClient.csproj:25]: yep it does :), suggest you fix it in this PR

Comment (nick5616) [reply src/UWPClientLibrary/Microsoft.FactoryOrchestrator.UWPClient.csproj:25]: Done!

**Review by nick5616 — COMMENTED** [review]
#168 Must be recommitted. This PR overwrites the changes found in PR #168

**Review by spaceisfun — APPROVED** [review]

**Review by patelp3989 — APPROVED** [review]

**Pull request merged**